src/docs/sdk/unified-api/index.mdx

```diff
@@ -73,136 +65,219 @@ meant that certain integrations (such as breadcrumbs) were often not possible.
+
+### `get_global_scope()`
+
+Returns the global scope, which is applied to all events dispatched by the SDK. This scope MUST be a singleton and shared across all instances of the SDK.
+
+### `last_event_id()`
+
+Returns the last event ID that was emitted. The last event ID SHOULD be kept on the isolation scope, and thus SHOULD be kept per isolated process. This is for instance used to implement user feedback dialogs.
+
+### `start_session()`
+
+Stores a session on the current scope and starts tracking it. This SHOULD attach a brand new session to the isolation scope, and implicitly ends any already existing session.
+
+### `end_session()`
+
+Ends the session, setting an appropriate `status` and `duration`, and enqueues it for sending to Sentry.
+
+## Concurrency
+
+All SDKs SHOULD have the concept of concurrency safe context storage, if the language permits it. What this means depends on the language. 
+
+The goal is that, wherever possible, concurrent operations can be automatically isolated from each other, in order to prevent data to leak between different operations.
+A classic example for this is to isolate requests in a HTTP server - if a user sets a user in this request, it should be applied to any event captured in this request, but not to events captured in other requests (because they may have a different user).
+
+The SDK SHOULD automatically manage this isolation, so that the user does not have to worry about it. For cases where the SDK cannot do that automatically, the SDK MUST provide a way for the user to manually manage the isolation (for example via `with_isolation_scope()`).
 
-The user can modify the current scope (to set extra, tags, current user) through the global function `configure_scope`.  `configure_scope` takes a callback function to which it passes the current scope.
+In client-side applications (Browser JS or mobile), there MAY not be a way to isolate code from each other. In this case, the SDK MAY fall back to handling state globally.
 
-The reason for this callback-based API is efficiency. If the SDK is disabled,
-it should not invoke the callback, thus avoiding unnecessary work.
+This is implemented as a thread local stack in most languages, or as a different construct (e.g. async local storage in Node.js) in others.
+
+### How to store concurrency data
+
+Each thread local SHOULD hold a reference to the current scope & isolation scope that is active for the current thread. 
+Newly spawned threads SHOULD inherit the reference to the isolation scope from the parent thread, and SHOULD create a new scope that is a child of the parent current scope, and make this the current scope for the child thread.
+
+The isolation scope MAY be forked (instead of using the same reference) for a new thread if the SDK identifies that the new thread is a new logical operation that should not share data with the parent thread. 
```

Comment:
can we have a small clarification that the current scope is mainly for span tracking purposes while the isolation scope is for tracking a unit of logical isolation